lib/rspec/rails/example/request_example_group.rb

```diff
@@ -1,5 +1,14 @@
 module RSpec
   module Rails
+    # Raises an error if the user tries to call integration test helpers.
+    # If the helpers are available then this module will be ignored.
+    module IntegrationHelperWarnings
+      extend ActiveSupport::Concern
+      [:assert_select, :url_for, :url_options].each do |name|
+        define_method(name) { |*args| raise "you must upgrade Rails to call #{name}" }
+      end
+    end
```

Comment:
I'm not sure we need this, we already get the `NoMethodError` for free after all ;)